utils/kubernetes/loader.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kubernetes
+
+import (
+	"encoding/json"
+	"fmt"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// LoadResourceFromYaml returns a Kubernetes resource from its serialized YAML definition.
+func LoadResourceFromYaml(scheme *runtime.Scheme, data string) (ctrl.Object, error) {
```

Comment:
this funcs are used to parse the sw definition from the crd, right?

Reply:
it is used to return a runtime Object (a CR instance) from a CR defined as a string.